File: python/maths/suiteRecurSinus.py
from math import sin, sqrt

def un(u0, n) :
    r = u0
    for i in range (n) :
        r = sin(r)
    return r

u0 = float(input("u0 = "))
n = int(input("n = "))
u_n = un(u0, n)
print(u_n)
print(u_n/sqrt(3/n))

File: ocaml/suiteRecurSinus.ml
let next un = sin(un);;

let un u0 n = 
    let r = ref u0 in
    for i=1 to n do
        r := next(!r);
    done;
    !r
;;

print_string "u0 = ";;
let u0 = read_float();;
print_string "n = ";;
let n = read_int();;
let r = un u0 n;;
print_string "u_n = ";;
print_int r;;
print_string "u_n/(3/sqrt n) = ";;
let rapport = sqrt(3. /. r);;